google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManager.java

```diff
@@ -18,19 +18,75 @@
 
 import com.google.api.core.BetaApi;
 import com.google.cloud.storage.BlobInfo;
-import java.io.IOException;
 import java.nio.file.Path;
 import java.util.List;
 import org.checkerframework.checker.nullness.qual.NonNull;
 
+/**
+ * An interface for a Transfer Manager.
+ *
+ * <p>Transfer Manager handles Parallel Uploads and Parallel Downloads.
+ */
 @BetaApi
 public interface TransferManager extends AutoCloseable {
 
+  /**
+   * Uploads a list of files in parallel.
+   *
+   * <p>Accepts a {@link ParallelUploadConfig} which defines the constraints of parallel uploads or
+   * predefined defaults.
+   *
+   * <p>Example of creating a parallel upload with Transfer Manager.
+   *
+   * <pre>{@code
+   * String bucketName = "my-unique-bucket";
+   * Path filePath = Paths.get("/path/to/my/file.txt");
+   * Path anotherFilePath = Paths.get("/path/to/another/file.txt");
+   * List<Path> files = List.of(filePath, anotherFilePath);
+   *
+   * ParallelUploadConfig parallelUploadConfig =
+   *           ParallelUploadConfig.newBuilder()
+   *               .setBucketName(bucketName)
+   *               .build();
+   *
+   * UploadJob uploadedFiles = transferManager.uploadFiles(files, config);
+   *
+   * }</pre>
+   *
+   * @return an {@link UploadJob}
```

Comment:
We should call out that this method will return quickly, and that awaiting results should be done on the UploadJob itself. Conversely, when we add the javadocs to UploadJob we should call out that getting results will block the invoking thread until the job is finished.

Same for DownloadJob